### PR TITLE
Demonstrate adding dynamic artwork in the splashscreen sample

### PR DESCRIPTION
**Pull request by ghost**
This uses a wxMemoryDC to show how to add dynamic art and text to the splashscreen. It's a nice way to spruce up this sample and probably has some real-world use.

What I am doing here is probably out of the scope of splashscreen sample, but I have this new code in a separate function and commented it (so that it's clear that this is optional).  Like I said, it makes this sample a little snazzier, but it's up to you if this seems appropriate.  Thanks for the consideration!

### CODE REVIEW

**Review by vadz — CHANGES_REQUESTED** [review]
I'm not really against adding this, although I do wonder if it doesn't distract from the purpose of the sample, which is to show wxSplashScreen class and not how to draw on the bitmaps (we have the drawing sample for this), but the code needs to be cleaned up a bit to at least show a good example of drawing.

File: samples/splash/splash.cpp [hunk 1]
```diff
@@ -141,6 +143,10 @@ bool MyApp::OnInit()
 
     if (ok)
     {
+        //we can even draw dynamic artwork onto our splashscreen
```
[the remark this answers is not in the document]
Comment (Blake-Eryx) [reply samples/splash/splash.cpp]: Fixed.

File: samples/splash/splash.cpp [hunk 2]
```diff
@@ -162,6 +168,41 @@ bool MyApp::OnInit()
...
+    memDc.SetTextForeground(*wxWHITE);
+    //calculate the height of the text so that we can nicely center it vertically
+    memDc.GetTextExtent(description, &width, &height);
+    memDc.DrawText(description, 10, bmp.GetHeight() - (height + 10));
+
+    memDc.SetFont(wxFontInfo(8));
+    memDc.GetTextExtent(copyrightLabel, &width, &height);
+    memDc.DrawText(copyrightLabel, bmp.GetWidth() - (width + 10), bmp.GetHeight() - (height + 10));
+
+    memDc.SelectObject(wxNullBitmap);
```
[the remark this answers is not in the document]
Comment (Blake-Eryx) [reply samples/splash/splash.cpp]: OK, I removed the SelectObject().

File: samples/splash/splash.cpp [hunk 3]
```diff
@@ -162,6 +168,41 @@ bool MyApp::OnInit()
...
 }
 
+//Draws artwork onto our splashscreen at runtime
+void MyApp::DecorateSplashScreen(wxBitmap& bmp)
+{
+    //use a memory DC to draw directly onto the bitmap
+    wxMemoryDC memDc(bmp);
+
+    //draw an orange box (with black outline) at the bottom of the splashscreen
+    wxDCPenChanger pc(memDc, wxPen(*wxBLACK_PEN));
```
[the remark this answers is not in the document]
Comment (Blake-Eryx) [reply samples/splash/splash.cpp]: Got it.

File: samples/splash/splash.cpp [hunk 4]
```diff
@@ -162,6 +168,41 @@ bool MyApp::OnInit()
...
+    wxDCBrushChanger bc(memDc, wxBrush(wxColour(255, 102, 0)));
+    memDc.DrawRectangle(wxRect(0, bmp.GetHeight() - 50, bmp.GetWidth(), bmp.GetHeight()));
+    memDc.DrawLine(0, bmp.GetHeight() - 50, bmp.GetWidth(), bmp.GetHeight() - 50);
+
+    //dynamically get the wxWidgets version to display
+    wxString description = wxString::Format("wxWidgets %s", wxVERSION_NUM_DOT_STRING);
+    //Create a copyright notice that uses the year that this file was compiled in
+    wxString year(__DATE__, *wxConvCurrent);
```
[the remark this answers is not in the document]
Comment (Blake-Eryx) [reply samples/splash/splash.cpp]: Pre version 2.9 code, must have been necessary back then. I removed it, thanks.

File: samples/splash/splash.cpp [hunk 5]
```diff
@@ -162,6 +168,41 @@ bool MyApp::OnInit()
...
+    wxString copyrightLabel = wxString::Format("%c%s wxWidgets. %s",
+        0xA9, year.Mid(year.Length() - 4),
```
[the remark this answers is not in the document]
Comment (Blake-Eryx) [reply samples/splash/splash.cpp]: Oh, cool, thanks.  I learned something new.  Fixed.

**Pull request closed without merging**